server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustLanguagePlugin.kt

```diff
@@ -0,0 +1,331 @@
+package org.jetbrains.bsp.bazel.server.sync.languages.rust
+
+import ch.epfl.scala.bsp4j.*
+import ch.epfl.scala.bsp4j.RustDependency
+import org.jetbrains.bsp.bazel.info.BspTargetInfo
+import org.jetbrains.bsp.bazel.info.BspTargetInfo.TargetInfo
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.model.Module
+import org.jetbrains.bsp.bazel.server.sync.languages.LanguagePlugin
+import org.jetbrains.bsp.bazel.info.BspTargetInfo.RustCrateLocation as ProtoRustCrateLocation
+import org.apache.logging.log4j.LogManager // TODO: remove
+import org.jetbrains.bsp.bazel.server.sync.model.Label
+import org.jetbrains.bsp.bazel.server.sync.model.Language
+
+class RustLanguagePlugin(private val bazelPathsResolver: BazelPathsResolver) : LanguagePlugin<RustModule>() {
+    override fun resolveModule(targetInfo: TargetInfo): RustModule? {
+        return targetInfo.getRustCrateInfoOrNull()?.run {
```

Comment:
a lot of nested things - it'd be nice to extract part of it to separate methods

Reply:
done, I guess